Makefile

```diff
@@ -94,7 +94,7 @@ test-pydocstyle:
 .PHONY: test-pylint
 test-pylint:
 	pylint rockcraft
-	pylint tests --disable=invalid-name,missing-module-docstring,missing-function-docstring,redefined-outer-name,too-many-arguments
+	pylint tests --disable=invalid-name,missing-module-docstring,missing-function-docstring,redefined-outer-name,too-many-arguments,protected-access
```

Comment:
I would remove protected-access here and make _run_in_provider public and add this explicitly to the callers that need it.

Reply:
Done